src/components/table/bl-table.ts

```diff
@@ -0,0 +1,202 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import "element-internals-polyfill";
+import { event, EventDispatcher } from "../../utilities/event";
+import style from "./bl-table.css";
+import BlTableCell from "./table-cell/bl-table-cell";
+import BlTableHeaderCell from "./table-header-cell/bl-table-header-cell";
+import BlTableRow from "./table-row/bl-table-row";
+
+export const blTableTag = "bl-table";
+
+export const blSortChangeEventName = "bl-table-sort";
+export const blRowSelectChangeEventName = "bl-table-row-select";
+
+/**
+ * @tag bl-table
+ * @summary Baklava Button component
+ *
+ */
+@customElement(blTableTag)
+export default class BlTable extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  /**
+   * Selected table row selection key list
+   */
+  @property({ type: Array, reflect: true, attribute: "select-value" })
+  get selectValue(): string[] {
+    return this._selectValue;
+  }
+  set selectValue(value: string[]) {
+    this._selectValue = value;
+    this.updateComplete.then(() => {
+      Array.from(this.querySelectorAll("bl-table-header-cell,bl-table-cell,bl-table-row")).map(
```

Comment:
Perhaps to avoid the conversion cost, we can use directly document.querySelectorAll("bl-table-header-cell,bl-table-cell,bl-table-row").forEach